src/main/java/com/amazonaws/gurureviewercli/adapter/ArtifactAdapter.java

```diff
@@ -40,22 +41,50 @@
     public static ScanMetaData zipAndUpload(final Configuration config,
                                             final Path tempDir,
                                             final Path repositoryDir,
-                                            final List<String> sourceDirs,
-                                            final List<String> buildDirs,
+                                            final List<Path> sourceDirs,
+                                            final List<Path> buildDirs,
                                             final String bucketName) throws IOException {
         try {
-            val sourceDirsAndGit = new ArrayList<String>(sourceDirs);
-            if (config.getBeforeCommit() != null && config.getAfterCommit() != null) {
-                // only add the git folder if a commit range is provided.
-                sourceDirsAndGit.add(repositoryDir.resolve(".git").toAbsolutePath().toString());
+            boolean scanVersionedFilesOnly = false;
+            if (config.getVersionedFiles() != null && !config.getVersionedFiles().isEmpty()) {
+                scanVersionedFilesOnly =
+                    !config.isInteractiveMode() ||
+                    config.getTextIO()
+                          .newBooleanInputReader()
+                          .withTrueInput("y")
+                          .withFalseInput("n")
+                          .read("Only analyze files under version control?");
+            }
+            final String sourceKey;
+            if (scanVersionedFilesOnly) {
+                val filesToScan = new ArrayList<Path>(ZipUtils.getFilesInDirectories(sourceDirs));
+                val totalFiles = filesToScan.size();
+                filesToScan.retainAll(config.getVersionedFiles()); // only keep versioned files.
+                val versionedFiles = filesToScan.size();
+                if (versionedFiles == 0) {
+                    Log.error(sourceDirs.toString());
+                    Log.error(config.getVersionedFiles().toString());
+                    throw new GuruCliException(ErrorCodes.GIT_EMPTY_DIFF,
```

Comment:
Should this be an exception or a logged warning?

Reply:
This should end the workflow because there's no point sending anything to CodeGuru and it was probably a mistake. These exceptions don't show stacktraces, they just lead to an error message that aborts the upload and scan.